core/ir/aarch64/codec.py

```diff
@@ -338,6 +338,39 @@ def generate_opcode_names(patterns):
           '#endif /* OPCODE_NAMES_H */']
     return '\n'.join(c) + '\n'
 
+def generate_opcode_opnd_pairs(patterns):
+    c = ['#ifndef OPCODE_OPND_PAIRS_H',
+         '#define OPCODE_OPND_PAIRS_H 1',
+         '',
+         '#include <stdint.h>',
+         '',
+         'typedef struct {',
+         '  uint32_t opcode;',
+         '  uint32_t opnd;',
+         '} opcode_opnd_pair_t;',
+         '',
+         'const opcode_opnd_pair_t fuzz_opcode_opnd_pairs[] = {']
+    # Exclude instructions with side-effects and branch instructions.
+    # Particularly,  exclude: i) Load/Stores (opcode: x1x0);
+    # ii) Branches, Exception Generating and System instructions (opcode: 101x);
+    # iii) SVE memory (opcode: 0010 and 1 for the most significant bit).
+    # The allowed instructions include all the Data Processing instructions (including
+    # the FP and SIMD ones) and the non-memory SVE instructions.
+    excluded_opcodes = re.compile('....1.0|...101.|1..0010')
```

Comment:
AIUI the MSB is only 1 for the SME (Scalable Matrix Extension) instructions.

Reply:
Thanks for taking a look.
For finding the opcodes to exclude I relied on the Armv8 documentation of the encodings (https://developer.arm.com/documentation/ddi0596/2021-06/Index-by-Encoding/SVE-encodings?lang=en). In this documentation, SVE encodings with MSB 1 are marked as SVE memory. I assume that the newly introduced SME in Armv9 would also involve memory ops. If there are non-memory-involving ops in there that could be included in the fuzz app, we can always relax this exclusion in the future.

Reply:
You are right, my bad, I was looking at old documentation :-/